File: C++/Baju.cpp
#include "Aksesoris.cpp"

// class baju, turunan dari aksesoris
class Baju : public Aksesoris
{
private:
    string untuk; // baju buat siapa (anjing, kucing, dll)
    string size;  // ukuran baju
    string merk;  // merk baju

public:
    // constructor default, awalnya kosong semua
    Baju() : Aksesoris()
    {
        this->untuk = "";
        this->size = "";
        this->merk = "";
    }

    // constructor pake parameter, langsung isi data baju
    Baju(string id, string nama_produk, int harga_produk, int stok_produk, string jenis, string bahan, string warna, string untuk, string size, string merk)
        : Aksesoris(id, nama_produk, harga_produk, stok_produk, jenis, bahan, warna) // panggil constructor aksesoris
    {
        this->untuk = untuk;
        this->size = size;
        this->merk = merk;
    }

    // buat ambil info baju buat siapa
    string get_untuk()
    {
        return this->untuk;
    }

    // buat ubah info baju buat siapa
    void set_untuk(string untuk)
    {
        this->untuk = untuk;
    }

    // buat ambil ukuran baju
    string get_size()
    {
        return this->size;
    }

    // buat ubah ukuran baju
    void set_size(string size)
    {
        this->size = size;
    }

    // buat ambil merk baju
    string get_merk()
    {
        return this->merk;
    }

    // buat ubah merk baju
    void set_merk(string merk)
    {
        this->merk = merk;
    }

    // buat nampilin info baju
    void display()
    {
        Aksesoris::display(); // panggil display dari aksesoris biar atribut umum tetep ditampilin
        cout << " | untuk: " << untuk
             << " | size: " << size
             << " | merk: " << merk << '\n';
    }

    // destruktor, otomatis jalan kalo objeknya dihapus
    ~Baju()
    {
    }
};

File: Java/Baju.java
// class baju ini anak dari aksesoris, jadi semua fitur aksesoris bisa dipake juga di sini
public class Baju extends Aksesoris {
    private String untuk;
    private String size;
    private String merk;

    // constructor default, kalo objek dibuat tapi ga dikasih data langsung
    public Baju() {
        this.untuk = "";
        this.size = "";
        this.merk = "";
    }

    // constructor yang pake parameter biar langsung bisa set data pas buat objek
    public Baju(String id, String namaProduk, int hargaProduk, int stokProduk, String jenis, String bahan, String warna, String untuk, String size, String merk) {
        super(id, namaProduk, hargaProduk, stokProduk, jenis, bahan, warna); // manggil constructor dari aksesoris
        this.untuk = untuk;
        this.size = size;
        this.merk = merk;
    }

    // getter buat ngambil nilai dari variabel untuk
    public String getUntuk() {
        return this.untuk;
    }

    // setter buat ngubah nilai dari variabel untuk
    public void setUntuk(String untuk) {
        this.untuk = untuk;
    }

    // getter buat ambil size
    public String getSize() {
        return this.size;
    }

    // setter buat ubah size
    public void setSize(String size) {
        this.size = size;
    }

    // getter buat ambil merk
    public String getMerk() {
        return this.merk;
    }

    // setter buat ubah merk
    public void setMerk(String merk) {
        this.merk = merk;
    }
}
